serverless/serverless
# Add tags to AWS APIGateway Stage

## What did you implement:

Closes #4644 

## How did you implement it:
Added a [`AWS::ApiGateway::Stage`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html) resource that add `Tags` to stage.

This also allow us to configure `TracingEnabled` (XRay for API Gateway), `CacheCluster***` and other properties in subsequent PRs.


## How can we verify it:

1. `npm install -g exoego/serverless#tagging-apigw`
1. `sls deploy` over
```yaml
provider:
  name: aws
  runtime: nodejs8.10
  stackTags:
    foo: bar
functions:
  hello:
    handler: handler.hello
    events:
      - http: GET /user
```
1. Verify that new stage `dev` is created and it has tags `foo: bar`.

## Todos:

- [x] Write documentation (N/A ?)

***Is this ready for review?:*** YES
***Is it a breaking change?:*** COULD BE (API GW stage name is updated every deployment)
